libs/platform/src/lib/components/form/collection-base.input.ts

```diff
@@ -75,4 +126,47 @@ export abstract class CollectionBaseInput extends BaseInput {
             return this.objectGet(obj[is[0]], is.slice(1));
         }
     }
+
+    /** set values of inline for each screen layout */
+    protected _setFieldLayout(): void {
+        try {
+            this._sIsInline = this.inlineLayout['S'] || false;
+            this._mdIsInline = this.inlineLayout['M'];
+            this._lgIsInline = this.inlineLayout['L'];
+            this._xlIsInline = this.inlineLayout['XL'];
+            this._updateLayout();
+        } catch (error) {
+            this._isInLineLayoutEnabled = false;
+        }
+
+        if (this._isInLineLayoutEnabled) {
+            fromEvent(window, 'resize')
+                .pipe(debounceTime(50), takeUntil(this._destroyed))
+                .subscribe(() => this._updateLayout());
+        }
+    }
+
+    /** @hidden */
+    private _updateLayout(): void {
+        const width = window.innerWidth;
+
+        // check if value has changed, then only assign new value.
+        if (width > 0 && width < RESPONSIVE_BREAKPOINTS['S'] && this.isInline !== this._sIsInline) {
+            this.isInline = this._sIsInline;
+        } else if (
+            width >= RESPONSIVE_BREAKPOINTS['S'] &&
+            width < RESPONSIVE_BREAKPOINTS['M'] &&
+            this.isInline !== this._mdIsInline
+        ) {
+            this.isInline = this._mdIsInline;
+        } else if (
+            width >= RESPONSIVE_BREAKPOINTS['M'] &&
+            width < RESPONSIVE_BREAKPOINTS['L'] &&
+            this.isInline !== this._lgIsInline
+        ) {
+            this.isInline = this._lgIsInline;
+        } else if (width >= RESPONSIVE_BREAKPOINTS['L'] && this.isInline !== this._xlIsInline) {
+            this.isInline = this._xlIsInline;
+        }
```

Comment:
Can we leverage this [cdk layout utility](https://material.angular.io/cdk/layout/overview) 
Looks like it matches perfectly to our needs.

Reply:
[predefined values](https://material.angular.io/cdk/layout/overview#predefined-breakpoints) here, mismatches with our media query values present in styles (https://github.com/SAP/fundamental-styles/blob/main/src/form-layout-grid.scss#L115).
Should we try to use it?

Reply:
This is just [the constant](https://github.com/angular/components/blob/master/src/cdk/layout/breakpoints.ts)
We have to define our own constant and use it.
Btw it would be great define RESPONSIVE_BREAKPOINTS as injectable constant so user is able to redefine it if needed

Reply:
using breakpointObserver from cdk.
also `BreakPoints` are injectable now, so custom values for different breakpoints like XL, L, M, S can be provided. default value is system default coming from Styles repo.